library/core/src/slice/raw.rs

```diff
@@ -83,6 +83,27 @@ use crate::ub_checks;
 /// }
 /// ```
 ///
+/// ### FFI: Handling null pointers
+///
+/// In languages such as C++, pointers to empty collections are not guaranteed to be non-null.
+/// When accepting such pointers, they have to be checked for null-ness to avoid undefined
+/// behavior.
+///
+/// ```
+/// use std::slice;
+///
+/// unsafe extern "C" fn handle_slice(ptr: *const f32, len: usize) {
+///     let data = if ptr.is_null() {
+///         // `len` is assumed to be 0.
+///         &[]
+///     } else {
+///         unsafe { slice::from_raw_parts(ptr, len) }
+///     };
+///     dbg!(data);
+///     // ...
+/// }
+/// ```
```

Comment:
Thanks for the review!

I have added the "safety" comments, please let me know if I should make any changes to the text.

I also changed the function to return a value and added an assert.

I'm not sure if that makes sense, though, since this function would normally be called from C++ or some other FFI language.

Alternatively, I could also provide a more general (and more useful) function:

```rust
unsafe fn ffi_slice<'a, T>(ptr: *const T, len: usize) -> &'a [T] {
    if ptr.is_null() {
        // NB: `len` is assumed to be 0.
        &[]
    } else {
        unsafe { std::slice::from_raw_parts(ptr, len) }
    }
}
```

I have actually used that function in my code: https://github.com/AudioSceneDescriptionFormat/asdfspline-rust/blob/e2608b07e2749caf71e43f23d4592849d27dd450/ffi/src/lib.rs#L59-L89.

However, this is a bit more complicated due to the lifetime annotation, so it might not be appropriate here?